sara/curso/week-7/CV/frontend/src/app/details/details.component.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/prefer-default-export */
+import { Component, OnInit, AfterViewInit } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { switchMap, tap } from 'rxjs/operators';
+import { HttpService } from '../services/http.service';
+
+@Component({
+  selector: 'app-details',
+  templateUrl: './details.component.html',
+  styleUrls: ['./details.component.scss'],
+})
+export class DetailsComponent implements OnInit, AfterViewInit {
+  refresh$ = new Subject();
+
+  newInfoArray:Array<any> = []
+
+  fetchCv$!: Observable<any>;
+
+  constructor(public HttpService: HttpService) {}
+
+  ngOnInit(): void {
+    this.fetchCv$ = this.refresh$
+      .pipe(
+        switchMap(() => this.HttpService.fechtCV()),
+      );
+  }
+
+  ngAfterViewInit() {
+    this.refresh$.next();
+  }
+
+  callHttpService(newInfo:object) {
+    this.HttpService.updateCV(newInfo)
+      .pipe(
+        tap(() => this.refresh$.next()),
+      )
+      .subscribe();
+  }
+
+  updateCvInfo(keyName:String, arrayToChange:Array<any>, i:number, newInfo:string) {
+    this.newInfoArray = arrayToChange;
+    this.newInfoArray[i] = newInfo;
+    switch (keyName) {
```

Comment:
I like how you use a 'switch instead of create different function for every case